### PR TITLE
Add login endpoint

### PR DESCRIPTION
**Pull request by CodeReaper**

### CODE REVIEW

File: domains/auth/Auth.API/API/Controllers/LoginController.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    [HttpGet()]
+    [Route("auth/login")]
+    public async Task<IActionResult> GetAsync(IDiscoveryCache discoveryCache, IOptions<OidcOptions> oidcOptions, ILogger<LoginController> logger)
+    {
+        var discoveryDocument = await discoveryCache.GetAsync();
+        if (discoveryDocument == null || discoveryDocument.IsError)
+        {
+            logger.LogError("Unable to fetch discovery document: {Error}", discoveryDocument?.Error);
+            var uri = new Uri(oidcOptions.Value.RedirectUri);
+            return RedirectPreserveMethod(uri.AddQueryParameters("errorCode", "2").ToString());
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Controllers/LoginController.cs:21]:
```suggestion
            return RedirectPreserveMethod(QueryHelpers.AddQueryString(uri.ToString(), "errorCode", "2"));
```

Comment (robertrosborg) [reply domains/auth/Auth.API/API/Controllers/LoginController.cs:21]: From Microsoft.AspNetCore.WebUtilities

File: domains/auth/Auth.API/API/Controllers/LoginController.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+using API.Options;
+using API.Utilities;
+using IdentityModel.Client;
+using Microsoft.AspNetCore.Mvc;
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Controllers/LoginController.cs:4]:
```suggestion
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.WebUtilities;
```

File: domains/auth/Auth.API/Tests/Controllers/LoginControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    [Fact]
+    public async Task GetAsync_ShouldReturnRedirectToOurselves_WhenDiscoveryCacheFails()
+    {
+        var authorityHost = "them.com";
+        var callbackHost = "us.com";
+        var clientId = Guid.NewGuid().ToString();
+        var document = DiscoveryDocument.Load(new List<KeyValuePair<string, string>>() { new("error", "it went all wrong") });
+
+        var cache = Mock.Of<IDiscoveryCache>();
```
Comment (robertrosborg) [inline domains/auth/Auth.API/Tests/Controllers/LoginControllerTests.cs:54]: This all seemed ok until I saw the SetValue on the private field on DiscoveryDocument. Shouldn't we create our own test subclass of IDiscoveryCache?

Comment (CodeReaper) [reply domains/auth/Auth.API/Tests/Controllers/LoginControllerTests.cs:54]: Subclassing `IDiscoveryCache` will not help us create an instance of `DiscoveryDocumentResponse` which is resisting getting mocked.

Comment (CodeReaper) [reply domains/auth/Auth.API/Tests/Controllers/LoginControllerTests.cs:54]: We had a session trying to create an instance of `DiscoveryDocumentResponse` properly, which ended with us deciding to postpone it in favour of completing this PR.

**Review by Frydensberg1 — APPROVED** [review]

**Pull request merged**